src/java.base/unix/classes/sun/nio/fs/UnixChannelFactory.java

```diff
@@ -236,20 +236,6 @@ protected static FileDescriptor open(int dfd,
         if (flags.direct)
             oflags |= O_DIRECT;
 
-        // permission check before we open the file
-        @SuppressWarnings("removal")
-        SecurityManager sm = System.getSecurityManager();
-        if (sm != null) {
-            if (pathForPermissionCheck == null)
-                pathForPermissionCheck = path.getPathForPermissionCheck();
-            if (flags.read)
-                sm.checkRead(pathForPermissionCheck);
-            if (flags.write)
-                sm.checkWrite(pathForPermissionCheck);
-            if (flags.deleteOnClose)
-                sm.checkDelete(pathForPermissionCheck);
-        }
-
```

Comment:
The changes in this area need to go a bit deeper and remove the pathForPermissionCheck parameter from this method, the open method, and all the callers.